File: server/src/models/ChecklistTemplate.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IChecklistTemplateItem {
  label: string;
  description?: string;
  required: boolean;
  order: number;
}

export interface IChecklistTemplate extends Document {
  name: string;
  description?: string;
  items: IChecklistTemplateItem[];
  createdAt: Date;
  updatedAt: Date;
}

const ChecklistTemplateSchema: Schema = new Schema(
  {
    name: { type: String, required: true },
    description: { type: String },
    items: [
      {
        label: { type: String, required: true },
        description: { type: String },
        required: { type: Boolean, default: true },
        order: { type: Number, required: true },
      },
    ],
  },
  { timestamps: true }
);

export default mongoose.model<IChecklistTemplate>('ChecklistTemplate', ChecklistTemplateSchema);

File: mobile/lib/models/checklist_template.dart
class ChecklistTemplate {
  final String id;
  final String name;
  final String? description;
  final List<ChecklistTemplateItem> items;

  ChecklistTemplate({
    required this.id,
    required this.name,
    this.description,
    required this.items,
  });

  factory ChecklistTemplate.fromJson(Map<String, dynamic> json) {
    return ChecklistTemplate(
      id: json['_id'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      description: json['description'],
      items: (json['items'] as List<dynamic>?)
              ?.map((item) => ChecklistTemplateItem.fromJson(item))
              .toList() ??
          [],
    );
  }
}

class ChecklistTemplateItem {
  final String label;
  final String? description;
  final bool required;
  final int order;

  ChecklistTemplateItem({
    required this.label,
    this.description,
    required this.required,
    required this.order,
  });

  factory ChecklistTemplateItem.fromJson(Map<String, dynamic> json) {
    return ChecklistTemplateItem(
      label: json['label'] ?? '',
      description: json['description'],
      required: json['required'] ?? true,
      order: json['order'] ?? 0,
    );
  }
}
